pkg/nsx/services/inventory/builder.go

```diff
@@ -400,7 +402,10 @@ func (s *InventoryService) BuildService(service *corev1.Service) (retry bool) {
 		}
 	}
 
-	if len(networkErrors) > 0 {
+	if len(podIDs) == 0 && !isAnnotationError && hasAddr {
+		status = InventoryStatusUnknown
+		netStatus = NetworkStatusUnhealthy
+	} else if len(networkErrors) > 0 {
 		status = InventoryStatusDown
 		netStatus = NetworkStatusUnhealthy
 	}
```

Comment:
Can we merge the conditions here and the ones in line 367-388?
 https://github.com/vmware-tanzu/nsx-operator/blob/cb819f4d69f1062f1a385127d4aa502890182a70/pkg/nsx/services/inventory/builder.go#L367-L388
I feel the current logic is complicated, under some condition the status is set back and forth. Could we have a clear summary on how the service status is determined?

Reply:
fixed